subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -405,9 +714,248 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
+	if (err) {
+		LOG_ERR("Failed to generate bond name (err %d)", err);
+		return;
+	}
+
+	err = settings_save_one(name, &bond.nv, sizeof(bond.nv));
+	if (err) {
+		LOG_ERR("Failed to save bond (err %d)", err);
+		return;
+	}
+
+	fp_bonds[bond_idx] = bond;
+}
+
+static int bonded_by_addr_find(size_t *bond_idx, const bt_addr_le_t *addr)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		if (fp_bonds[i].bonded &&
+		    bt_addr_le_eq(&fp_bonds[i].nv.addr, addr)) {
+			*bond_idx = i;
+			return 0;
+		}
+	}
+
+	return -ESRCH;
+}
+
+void fp_storage_ak_bond_bt_request_cb_register(const struct fp_storage_ak_bond_bt_request_cb *cb)
+{
+	__ASSERT_NO_MSG(IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_BOND_MANAGEMENT));
+	__ASSERT_NO_MSG(cb);
+
+	bt_request_cb = cb;
+}
+
+int fp_storage_ak_bond_save(const bt_addr_le_t *addr, const void *conn_ctx,
+			    const struct fp_account_key *account_key)
+{
+	__ASSERT_NO_MSG(IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_BOND_MANAGEMENT));
+
+	if (!is_enabled) {
+		return -EACCES;
+	}
+
+	int err;
+	uint8_t id;
+	size_t bond_idx;
+	struct fp_bond bond;
+	char name[SETTINGS_BOND_NAME_MAX_SIZE];
+
+	if (account_key) {
+		err = ak_metadata_id_get(&id, account_key);
+		if (err) {
+			return err;
+		}
+	} else {
+		/* The actual ID will be determined during Account Key write. */
+		id = 0;
+	}
+
+	err = bond_free_idx_get(&bond_idx);
+	if (err) {
+		return err;
+	}
+
+	bond.conn_ctx = conn_ctx;
+	bond.bonded = false;
+	bond.nv.account_key_metadata_id = id;
+	bond.nv.addr = *addr;
+
+	err = bond_name_gen(name, bond_idx);
+	if (err) {
+		return err;
+	}
+
+	err = settings_save_one(name, &bond.nv, sizeof(bond.nv));
+	if (err) {
+		return err;
+	}
+
+	fp_bonds[bond_idx] = bond;
+
+	return 0;
+}
+
+void fp_storage_ak_bond_confirm(const bt_addr_le_t *addr, const void *conn_ctx)
+{
+	__ASSERT_NO_MSG(IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_BOND_MANAGEMENT));
+
+	if (!is_enabled) {
+		return;
+	}
+
+	size_t bond_idx;
+	size_t bond_duplicate_idx;
+	int duplicate_find_err = 0;
+	int err;
+
+	err = bond_find(&bond_idx, conn_ctx);
+	if (err) {
+		return;
+	}
+
+	fp_bonds[bond_idx].bonded = true;
+
+	while (!duplicate_find_err) {
+		duplicate_find_err = bonded_duplicate_find(&bond_duplicate_idx, addr, conn_ctx);
+		if (!duplicate_find_err) {
+			err = bond_delete(bond_duplicate_idx);
+			if (err) {
+				LOG_ERR("Failed to delete bond (err %d)", err);
+				break;
+			} else {
+				LOG_DBG("Deleted duplicated bond");
+			}
+		}
+	}
+
+	if (fp_bonds[bond_idx].nv.account_key_metadata_id) {
+		/* Account Key already associated (subsequent pairing procedure).
+		 * Procedure finished successfully. Setting conn_ctx to NULL.
+		 */
+		fp_bonds[bond_idx].conn_ctx = NULL;
+	}
+}
+
+void fp_storage_ak_bond_update(const void *conn_ctx, const bt_addr_le_t *new_addr)
+{
+	__ASSERT_NO_MSG(IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_BOND_MANAGEMENT));
+
+	if (!is_enabled) {
+		return;
+	}
+
+	bond_update_addr(new_addr, conn_ctx);
```

Comment:
We could directly write to field representing address of proper conn and trigger NVM store here